src/Integration.UnitTests/SonarAnalyzer/ActiveSolutionBoundTrackerTests.cs

```diff
@@ -86,40 +84,32 @@ public void ActiveSolutionBoundTracker_Bound()
         public void ActiveSolutionBoundTracker_Changes()
         {
             // Setup
-            var solutionBinding = new ConfigurableSolutionBindingSerializer
-            {
-                CurrentBinding = new BoundSonarQubeProject()
-            };
-            this.serviceProvider.RegisterService(typeof(ISolutionBindingSerializer), solutionBinding);
+            this.solutionBindingInformationProvider.BoundProjects = new[] { new ProjectMock("") };
             var testSubject = new ActiveSolutionBoundTracker(this.host, this.activeSolutionTracker);
+            var reanalysisEventCalledCount = 0;
+            testSubject.SolutionBindingChanged += (obj, args) => { reanalysisEventCalledCount++; };
 
             // Sanity
             Assert.IsTrue(testSubject.IsActiveSolutionBound, "Initially bound");
+            Assert.AreEqual(0, reanalysisEventCalledCount, "No reanalysis forced");
 
             // Case 1: Clear bound project
             // Act
-            solutionBinding.CurrentBinding = null;
-            host.VisualStateManager.ClearBoundProject();
+            this.solutionBindingInformationProvider.BoundProjects = Enumerable.Empty<EnvDTE.Project>();
+            this.host.VisualStateManager.ClearBoundProject();
 
             // Verify
             Assert.IsFalse(testSubject.IsActiveSolutionBound, "Unbound solution should report false activation");
+            Assert.AreEqual(1, reanalysisEventCalledCount, "Unbind should trigger reanalysis");
 
             // Case 2: Set bound project
-            solutionBinding.CurrentBinding = new BoundSonarQubeProject();
+            this.solutionBindingInformationProvider.BoundProjects = new[] { new ProjectMock("") };
             // Act
-            host.VisualStateManager.SetBoundProject(new ProjectInformation());
+            this.host.VisualStateManager.SetBoundProject(new ProjectInformation());
 
             // Verify
             Assert.IsTrue(testSubject.IsActiveSolutionBound, "Bound solution should report true activation");
-
-            // Case 3: Dispose and change
```

Comment:
Any reason why this is gone? It was checking that the events are unregistered in order not to leak them